### PR TITLE
[ASP-4095] Improve error handling and reporting in Jobbergate

### PR DESCRIPTION
**Pull request by fschuch**
#### What
* Reviewed error messages from the API side
* Added back syntax check to upload files to provide early feeddback on bad data
* Modified the CLI to present more descriptive error messages and possible solution for comum problems

#### Why
As a Jobbergate user, I want more descriptive error messages, so that I can better understand the problem and better express it when requiring support.

`Task`: https://jira.scania.com/browse/ASP-4095

---

#### Peer Review
Please follow the upstream omnivector documentation concerning
[peer-review guidelines](https://github.com/omnivector-solutions/Documentation/blob/main/Contributing/pr_review_standards.md#peer-review).

### CODE REVIEW

File: jobbergate-api/jobbergate_api/apps/file_validation.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+        validator = syntax_validation_dispatch[suffix]
+        result = validator(file_obj.read())
+        file_obj.seek(0)
+        return result
+    logger.debug(f"Skipping because {suffix} has no syntax validation")
+    return True
+
+
+SyntaxValidationEquation = Callable[[Union[str, bytes]], bool]
+"""Type alias describing the function signature used to validate file syntax."""
```
Comment (dusktreader) [inline jobbergate-api/jobbergate_api/apps/file_validation.py:38]: Why not use a normal comment (using `#`) here?

File: jobbergate-api/jobbergate_api/apps/job_scripts/routers.py [hunk 2]
```diff
@@ -97,7 +99,9 @@ async def job_script_create_from_template(
     if len(missing_keys) > 0:
         raise HTTPException(
             status_code=status.HTTP_400_BAD_REQUEST,
-            detail=(f"The template files are missing required files: {missing_keys} for {id_or_identifier}"),
+            detail=("The following required files are missing for job template {}: {}").format(
```
Comment (dusktreader) [inline jobbergate-api/jobbergate_api/apps/job_scripts/routers.py:102]: I'm wondering if the extra parens are necessary?

```suggestion
            detail="The following required files are missing for job template {}: {}".format(
```

**Review by dusktreader — APPROVED** [review]
Very nice work! Added couple of trivial comments that you are free to ignore if you wish.

File: jobbergate-api/jobbergate_api/apps/file_validation.py (hunk 1 above)
Comment (fschuch) [reply jobbergate-api/jobbergate_api/apps/file_validation.py:38]: Good catch, solved in [c06c77a](https://github.com/omnivector-solutions/jobbergate/pull/450/commits/c06c77a365b850f51b4a65abe680dcfac9c2cce9)

File: jobbergate-api/jobbergate_api/apps/job_scripts/routers.py (hunk 2 above)
Comment (fschuch) [reply jobbergate-api/jobbergate_api/apps/job_scripts/routers.py]: Good catch, solved in [c06c77a](https://github.com/omnivector-solutions/jobbergate/pull/450/commits/c06c77a365b850f51b4a65abe680dcfac9c2cce9)

**Pull request merged**